resources/sdk/purecloudjava/templates/Logger.mustache

```diff
@@ -0,0 +1,407 @@
+            this.statusCode = statusCode;
+            this.requestBody = requestBody;
+            this.responseBody = responseBody;
+        }
+    
+        // trace constructor
+        LogStatement(
+            Date date,
+            String level,
+            String method,
+            String url,
+            Map<String, String> requestHeaders,
+            Map<String, String> responseHeaders,
+            int statusCode,
+            String requestBody
+        ) {
+            this.date = date;
+            this.level = level;
+            this.method = method;
+            this.url = url;
+            this.requestHeaders = requestHeaders;
+            this.responseHeaders = responseHeaders;
+            this.correlationId = getCorrelationId(responseHeaders);
+            this.statusCode = statusCode;
+            this.requestBody = requestBody;
+        }
+
+        // debug constructor
+        LogStatement(
+            Date date,
+            String level,
+            String method,
+            String url,
+            Map<String, String> requestHeaders,
+            int statusCode,
+            String requestBody
+        ) {
+            this.date = date;
+            this.level = level;
+            this.method = method;
+            this.url = url;
+            this.requestHeaders = requestHeaders;
+            this.statusCode = statusCode;
+            this.requestBody = requestBody;
+        }
+
+        Date getDate() {
+            return date;
+        }
+
+        String getLevel() {
+            return level;
+        }
+
+        String getMethod() {
+            return method;
+        }
+
+        String getUrl() {
+            return url;
+        }
+
+        Map<String, String> getRequestHeaders() {
+            return requestHeaders;
+        }
+
+        Map<String, String> getResponseHeaders() {
+            return responseHeaders;
+        }
+
+        int getStatusCode() {
+            return statusCode;
+        }
+
+        String getRequestBody() {
+            return requestBody;
+        }
+
+        String getResponseBody() {
+            return responseBody;
+        }
+    
+        String asString(LogFormat logFormat, boolean logRequestBody, boolean logResponseBody) {
+            this.requestHeaders.put("Authorization", "[REDACTED]");
+            if (!logRequestBody)
+                this.requestBody = null;
+            if (!logResponseBody)
+                this.responseBody = null;
+            if (logFormat == LogFormat.JSON) {
+                ObjectMapper objectMapper = new ObjectMapper();
+                objectMapper.setDateFormat(new SimpleDateFormat("yyyy-MM-dd HH:mm:ss"));
+                objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+                try {
+                    return objectMapper.writeValueAsString(this);
+                } catch (Exception e) {
+                    e.printStackTrace();
```

Comment:
Shouldn't print this. Also, for me it's throwing an exception telling me to add the following line:
`objectMapper.disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);`
When I add that it prints an empty JSON object.
The API I was calling is `usersApi.getUsersMe(null, null);`

Reply:
Also it seems to be hitting this code despite logFormat being set to Text and successfully logging text to the console and file

Reply:
And it's printing text when logFormat is set to JSON

Reply:
This turned out to be caused by changing the access level of the getters to "default".  I set them back to public and seeing the logs in json format again.